packages/hic/src/HicTrack/model.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  ConfigurationReference,
+  getConf,
+} from '@gmod/jbrowse-core/configuration'
+import { getParentRenderProps } from '@gmod/jbrowse-core/util/tracks'
+import { getSession } from '@gmod/jbrowse-core/util'
+import { blockBasedTrackModel } from '@gmod/jbrowse-plugin-linear-genome-view'
+import { types } from 'mobx-state-tree'
+import copy from 'copy-to-clipboard'
+import PluginManager from '@gmod/jbrowse-core/PluginManager'
+import { Feature } from '@gmod/jbrowse-core/util/simpleFeature'
+import { PileupConfigModel } from './configSchema'
+
+// using a map because it preserves order
+const rendererTypes = new Map([
+  ['pileup', 'PileupRenderer'],
+  ['svg', 'SvgFeatureRenderer'],
+  ['snpcoverage', 'SNPCoverageRenderer'],
+])
+
+export default (
+  pluginManager: PluginManager,
+  configSchema: PileupConfigModel,
+) =>
+  types
+    .compose(
+      'PileupTrack',
+      blockBasedTrackModel,
+      types.model({
+        type: types.literal('PileupTrack'),
+        configuration: ConfigurationReference(configSchema),
+      }),
+    )
+    .actions(self => ({
+      selectFeature(feature: Feature) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const session = getSession(self) as any
```

Comment:
can use `SessionWithDrawerWidgets` and the `isSessionWithDrawerWidgets` type guard here